lib/Sema/TypeCheckSwitchStmt.cpp

```diff
@@ -1344,18 +1345,28 @@ namespace {
       bool InEditor = TC.Context.LangOpts.DiagnosticsEditorMode;
 
       // Decide whether we want an error or a warning.
-      auto mainDiagType = diag::non_exhaustive_switch;
+      bool mainDiagnosticIsJustAWarning = false;
+
+      auto mainDiagnosticType = [&]() -> swift::Diagnostic {
+        if (defaultReason == RequiresDefault::SpaceTooLarge)
```

Comment:
I don't think you need all this complexity. `SpaceTooLarge` always produces an error, so you can just switch it to produce your new error instead.

I also really don't like this "set a variable captured by a lambda to change its behavior" business. If we *did* need this lambda, I'd suggest just making the "is warning" flag an argument rather than a capture.